accounts/views.py

```diff
@@ -172,3 +178,47 @@ class LabsProtectedViewSet(LabsView):
 
     def get(self, request, format=None):
         return HttpResponse({"secret_information": "this is a Penn Labs protected route"})
+
+
+class ProductAdminView(APIView):
+    """
+    Idempotently set admin permissions on all of our products.
+    Takes in a POST body in the form {"pennkey": ["permissions"]}
+    """
+
+    permission_classes = [HasAPIKey]
+
+    def post(self, request, format=None):
+        # Revoke all existing admin permissions
+        content_type = ContentType.objects.get(app_label="accounts", model="user")
+        perms = Permission.objects.filter(content_type=content_type, codename__endswith="_admin")
+        for perm in perms:
+            perm.user_set.clear()
+        User.objects.filter(Q(is_superuser=True) | Q(is_staff=True)).update(
+            is_superuser=False, is_staff=False
+        )
+
+        try:
+            body = json.loads(request.body)
+        except JSONDecodeError:
+            return HttpResponseBadRequest()
+
+        for pennkey in body:
+            user = User.objects.filter(username=pennkey).first()
+            if user is not None:
+                for permission_slug in body[pennkey]:
+                    # Handle platform separately
+                    if permission_slug == "platform_admin":
+                        user.is_superuser = True
+                        user.is_staff = True
+                        user.save()
+                        continue
+
+                    permission_name = f"{permission_slug[:-6]} Admin"
```

Comment:
This permission name wouldn't be formatted as our others are formatted... the name for the clubs permission for example is Clubs Admin , and since this receives penn_clubs_admin it would create the name as penn_clubs Admin

Reply:
Yeah, I got a bit lazy here, the name of the permission isn't parsed anywhere, so I did the minimum to make sure we could identify the permission and then update the name manually. Just pushed a fix to do that automatically